### PR TITLE
Rename every instance of "OGG" to "Ogg"

### LINKED ISSUE #4675: Rename every "OGG" statement to "Ogg"
### Describe the project you are working on

A game that has music.

### Describe the problem or limitation you are having in your project

Ogg is not an abbreviation, it is officially Ogg according to Xiph.org itself, therefore it shouldn't be written as OGG despite many web articles describing it as such, probably because of how WAV is all caps.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Rename every instance of OGG to Ogg.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Rename the following classes:
```
AudioStreamOGGVorbis
OGGPacketSequence
AudioStreamPlaybackOGGVorbis
OGGPacketSequencePlayback
```

To:

```
AudioStreamOggVorbis
OggPacketSequence
AudioStreamPlaybackOggVorbis
OggPacketSequencePlayback
```

I was considering the Ogg in the Vorbis class names should be omitted, but since its extension is `.ogg` 99% of the time, keeping it is helpful.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

We're attempting to rename a built-in class.

### PR DESCRIPTION
**Pull request by DeeJayLSP**
Implements godotengine/godot-proposals#4675

Every instance has been carefully reviewed. I built for testing, and everything seemed to work fine.

I made the PR since the proposal got many upvotes in a reasonable amount of time.

_Closes https://github.com/godotengine/godot-proposals/issues/4675_

### CODE REVIEW

**Review by Calinou — DISMISSED** [review]
Compatibility handlers should be added to rename existing nodes/resources from older projects: https://github.com/godotengine/godot/blob/f41cb30f9b7fe46a014162aba9c0e9a9697343d2/scene/register_scene_types.cpp#L935-L936

IIRC, this is not needed for classes that are neither nodes nor resources or aren't exposed to the scripting API.

**Comment by DeeJayLSP** [comment]
Do I need to do it for OGGPacketSequence? Since it's introduced in 4.0 I don't feel the need to add compatibility between alphas. If it's not needed, I will be excluding that line.

**Comment by Calinou** [comment]
> Does it need compatibility for OGGPacketSequence? Since it's introduced in 4.0 I don't feel the need to add compatibility between alphas. If it's not needed, I will be excluding that line.

I would add a rename for it: https://github.com/godotengine/godot/blob/4366f8bcd455714a5cd29374726d83b674ff0430/scene/register_scene_types.cpp#L1068

**Comment by DeeJayLSP** [comment]
> > Does it need compatibility for OGGPacketSequence? Since it's introduced in 4.0 I don't feel the need to add compatibility between alphas. If it's not needed, I will be excluding that line.
> 
> I would add a rename for it, as we intend to make the transition between various 4.0 alphas more seamless:
> 
> https://github.com/godotengine/godot/blob/4366f8bcd455714a5cd29374726d83b674ff0430/scene/register_scene_types.cpp#L1068

Then my pull request here is done.

**Comment by DeeJayLSP** [comment]
Moved the OGGPacketSequence one to the alpha rename list.

**Comment by DeeJayLSP** [comment]
All done.

**Comment by fire** [comment]
Since prs with a lot of changes are fragile. Will look into seeing how to merge this sooner.

**Comment by DeeJayLSP** [comment]
> Edited: Not that many instances of change.

All changes could be summarized into:
- All "OGG"s were renamed into "Ogg";
- Compatibility handlers for AudioStreamOGGVorbis and OGGPacketSequence added.

That's all.

**Comment by akien-mga** [comment]
Should also be good to merge once rebased (might be easy to redo from scratch and force push the new commit, that solving all merge conflicts).

**Review by akien-mga — APPROVED** [review]
OK to merge once rebased.

**Comment by DeeJayLSP** [comment]
> OK to merge once rebased.

@akien-mga Yuri said before to wait for #62896. Should I do it now?

**Comment by akien-mga** [comment]
> > OK to merge once rebased.
> 
> @akien-mga Yuri said before to wait for #62896. Should I do it now?

Yes this PR won't be merged as is, it will be reworked for 4.1. Some relevant parts have been moved to https://github.com/godotengine/godot/pull/63265 which got merged (which is likely the one causing the merge conflicts you now have).

File: editor/project_converter_3_to_4.cpp [hunk 1]
```diff
@@ -1252,6 +1252,7 @@ static const char *class_renames[][2] = {
 	{ "Area", "Area3D" }, // Be careful, this will be used everywhere
 	{ "AudioStreamRandomPitch", "AudioStreamRandomizer" },
 	{ "AudioStreamSample", "AudioStreamWAV" },
+	{ "AudioStreamOGGVorbis", "AudioStreamOggVorbis" },
```
Comment (akien-mga) [inline editor/project_converter_3_to_4.cpp:1255]: Should be alphabetically ordered.

File: doc/classes/AudioStream.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
 		Base class for audio streams.
 	</brief_description>
 	<description>
-		Base class for audio streams. Audio streams are used for sound effects and music playback, and support WAV (via [AudioStreamWAV]) and OGG (via [AudioStreamOGGVorbis]) file formats.
+		Base class for audio streams. Audio streams are used for sound effects and music playback, and support WAV (via [AudioStreamWAV]) and OGG (via [AudioStreamOggVorbis]) file formats.
```
Comment (akien-mga) [inline doc/classes/AudioStream.xml:7]:
```suggestion
		Base class for audio streams. Audio streams are used for sound effects and music playback, and support WAV (via [AudioStreamWAV]) and Ogg (via [AudioStreamOggVorbis]) file formats.
```

**Comment by akien-mga** [comment]
You missed some:
```
modules/vorbis/doc_classes/AudioStreamOggVorbis.xml
23:                     Contains the raw OGG data for this stream.

modules/vorbis/audio_stream_ogg_vorbis.cpp
388:Ref<AudioStreamPlayback> AudioStreamOGGVorbis::instantiate_playback() {
394:    ovs->vorbis_stream = Ref<AudioStreamOGGVorbis>(this);
406:String AudioStreamOGGVorbis::get_stream_name() const {
410:void AudioStreamOGGVorbis::maybe_update_info() {
446:void AudioStreamOGGVorbis::set_packet_sequence(Ref<OggPacketSequence> p_packet_sequence) {
453:Ref<OggPacketSequence> AudioStreamOGGVorbis::get_packet_sequence() const {
457:void AudioStreamOGGVorbis::set_loop(bool p_enable) {
461:bool AudioStreamOGGVorbis::has_loop() const {
465:void AudioStreamOGGVorbis::set_loop_offset(float p_seconds) {
469:float AudioStreamOGGVorbis::get_loop_offset() const {
473:float AudioStreamOGGVorbis::get_length() const {
478:void AudioStreamOGGVorbis::set_bpm(double p_bpm) {
484:double AudioStreamOGGVorbis::get_bpm() const {
488:void AudioStreamOGGVorbis::set_beat_count(int p_beat_count) {
494:int AudioStreamOGGVorbis::get_beat_count() const {
498:void AudioStreamOGGVorbis::set_bar_beats(int p_bar_beats) {
504:int AudioStreamOGGVorbis::get_bar_beats() const {
508:bool AudioStreamOGGVorbis::is_monophonic() const {
512:void AudioStreamOGGVorbis::_bind_methods() {
513:    ClassDB::bind_method(D_METHOD("set_packet_sequence", "packet_sequence"), &AudioStreamOGGVorbis::set_packet_sequence);
514:    ClassDB::bind_method(D_METHOD("get_packet_sequence"), &AudioStreamOGGVorbis::get_packet_sequence);
516:    ClassDB::bind_method(D_METHOD("set_loop", "enable"), &AudioStreamOGGVorbis::set_loop);
517:    ClassDB::bind_method(D_METHOD("has_loop"), &AudioStreamOGGVorbis::has_loop);
519:    ClassDB::bind_method(D_METHOD("set_loop_offset", "seconds"), &AudioStreamOGGVorbis::set_loop_offset);
520:    ClassDB::bind_method(D_METHOD("get_loop_offset"), &AudioStreamOGGVorbis::get_loop_offset);
522:    ClassDB::bind_method(D_METHOD("set_bpm", "bpm"), &AudioStreamOGGVorbis::set_bpm);
523:    ClassDB::bind_method(D_METHOD("get_bpm"), &AudioStreamOGGVorbis::get_bpm);
525:    ClassDB::bind_method(D_METHOD("set_beat_count", "count"), &AudioStreamOGGVorbis::set_beat_count);
526:    ClassDB::bind_method(D_METHOD("get_beat_count"), &AudioStreamOGGVorbis::get_beat_count);
528:    ClassDB::bind_method(D_METHOD("set_bar_beats", "count"), &AudioStreamOGGVorbis::set_bar_beats);
529:    ClassDB::bind_method(D_METHOD("get_bar_beats"), &AudioStreamOGGVorbis::get_bar_beats);
539:AudioStreamOGGVorbis::AudioStreamOGGVorbis() {}
541:AudioStreamOGGVorbis::~AudioStreamOGGVorbis() {}

modules/vorbis/resource_importer_ogg_vorbis.cpp
98:     Ref<OGGPacketSequence> ogg_packet_sequence;
140:                            ERR_FAIL_V_MSG(Ref<AudioStreamOggVorbis>(), "Failed allocating memory for OGG Vorbis stream.");
189:            ERR_FAIL_V_MSG(Ref<AudioStreamOggVorbis>(), "OGG Vorbis decoding failed. Check that your data is a valid OGG Vorbis audio stream.");

modules/vorbis/register_types.cpp
43:             Ref<ResourceImporterOGGVorbis> ogg_vorbis_importer;
49:     GDREGISTER_CLASS(AudioStreamPlaybackOGGVorbis);
```

**Comment by DeeJayLSP** [comment]
Not having your usual finder is painful.

**Comment by akien-mga** [comment]
Still missing a few, see my above review comments, and those in the error message:
```
modules/vorbis/resource_importer_ogg_vorbis.cpp
140:                            ERR_FAIL_V_MSG(Ref<AudioStreamOggVorbis>(), "Failed allocating memory for OGG Vorbis stream.");
189:            ERR_FAIL_V_MSG(Ref<AudioStreamOggVorbis>(), "OGG Vorbis decoding failed. Check that your data is a valid OGG Vorbis audio stream.");
```

**Comment by DeeJayLSP** [comment]
Also fixed a wrong rename

**Comment by akien-mga** [comment]
You're still missing my review comments:

https://github.com/godotengine/godot/pull/62164#discussion_r932570211

https://github.com/godotengine/godot/pull/62164#discussion_r932570395

The rest seems good.

**Comment by DeeJayLSP** [comment]
Honestly, it's ok to me if the AudioStream docs remain OGG, as it's mostly pointing to the file extension.

Still, it's better if the names remain consistent.

**Review by akien-mga — APPROVED** [review]

**Comment by DeeJayLSP** [comment]
For some reason, both this and the WAV one aren't showing icons for me.

It did show on files however, until I reimported them 😞.

**Comment by akien-mga** [comment]
That's probably a cache issue. A clean build should fix it.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**